tests/Fixer/AbstractFixerTest.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Fixer\Whitespace;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\Tests\Test\AbstractFixerTestCase;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @internal
+ *
+ * @covers \PhpCsFixer\Fixer\AbstractFixer
+ */
+final class AbstractFixerTest extends AbstractFixerTestCase
+{
+    public function testConfigureUnconfigurable()
+    {
+        $fixer = new UnconfigurableFixer();
+
+        $this->expectException('LogicException');
+        $this->expectExceptionMessage('Cannot configure using Abstract parent, child not implementing "PhpCsFixer\Fixer\ConfigurationDefinitionFixerInterface".');
+
+        $fixer->configure(array('foo' => 'bar'));
+    }
+
+    public function testGetConfigurationDefinitionUnconfigurable()
+    {
+        $fixer = new UnconfigurableFixer();
+
+        $this->expectException('LogicException');
+        $this->expectExceptionMessage('Cannot get configuration definition using Abstract parent, child not implementing "PhpCsFixer\Fixer\ConfigurationDefinitionFixerInterface".');
+
+        $fixer->getConfigurationDefinition();
+    }
+
+    public function testCreateConfigurationDefinitionUnconfigurable()
+    {
+        $fixer = new UnconfigurableFixer();
+
+        $this->expectException('LogicException');
+        $this->expectExceptionMessage('Cannot create configuration definition using Abstract parent, child not implementing "PhpCsFixer\Fixer\ConfigurationDefinitionFixerInterface".');
+
+        $fixer->doSomethingWithCreateConfigDefinition();
+    }
+
+    protected function createFixer()
+    {
+    }
+}
+
+final class UnconfigurableFixer extends AbstractFixer
```

Comment:
Any better way of doing this?

Reply:
This is not PSR-4 compliant, this class should be extracted to its own file.

Reply:
on 2.2 this looks good to me :+1: 
when we bump to PHP7 we could use an anonymous class as replacement for this test class

Reply:
This is not PSR-4 compliant, this class should be extracted to its own file.

Reply:
Which directory do you think it should go in?

Reply:
tests / fixtures

Reply:
Done